pkg/disttask/framework/scheduler/slots.go

```diff
@@ -196,3 +209,20 @@ func (sm *slotManager) updateCapacity(cpuCount int) {
 		}
 	}
 }
+
+func filterNodesWithEnoughSlots(usedSlots map[string]int, capacity int, eligibleNodes []string, concurrency int) []string {
+	nodesOfEnoughSlots := make(map[string]struct{}, len(usedSlots))
+	for n, slots := range usedSlots {
+		if slots+concurrency <= capacity {
+			nodesOfEnoughSlots[n] = struct{}{}
+		}
+	}
+
+	result := make([]string, 0, len(eligibleNodes))
+	for _, n := range eligibleNodes {
+		if _, ok := nodesOfEnoughSlots[n]; ok {
+			result = append(result, n)
+		}
+	}
```

Comment:
more meaningful naming

```suggestion
	for node, slots := range usedSlots {
		if slots+concurrency <= capacity {
			nodesOfEnoughSlots[node] = struct{}{}
		}
	}

	result := make([]string, 0, len(eligibleNodes))
	for _, node := range eligibleNodes {
		if _, ok := nodesOfEnoughSlots[node]; ok {
			result = append(result, node)
		}
	}
```